### ISSUE TITLE
Dark mode

### ISSUE THREAD

**Issue by AEsmerio** [buefy/buefy#1861]
I've searching a lot how people have implemented a Dark Theme / Dark Mode but no success so far. 
I'm planing to move my own project from Vuetify to Buefy, but Dark theme is required.
On Vuetify you only need set a prop in the component and it's done. So I can toggle the entire app from Light to Dark.

Am I right to say that there's no such thing as Dark mode property for the components?

**Comment by service-paradis** [comment]
You're right. There is no specific dark mode in Buefy right now.
You can customize the Bulma Settings to create your own dark theme. There are some interesting ones at https://jenil.github.io/bulmaswatch/

**Comment by jtommy** [comment]
@3rdSenna Bulma author is working on it, stay tuned!!

**Comment by franzvezuli** [comment]
> @3rdSenna Bulma author is working on it, stay tuned!!

Nice, great news! Is it tracked in a version or branch somewhere? Care to link? Want to watch it.

**Comment by WaldemarEnns** [comment]
I am waiting for it too :) Just like @3rdSenna , i saw it already on Vuetify and would love to have this feature in Buefy too! 🚀

**Comment by nicolaipre** [comment]
Would absolutely be nice to see a builtin dark mode implemented :-)

**Comment by a-kriya** [comment]
Perhaps this issue can be closed as it is a duplicate of #2076. Once the PR for it is merged, the variables can be configured as documented here https://bulma.io/2019/10/15/light-dark-colors/#scheme-variables-for-dark-mode.

**Comment by jtommy** [comment]
@a-kriya I don't think we'll able to get a dark mode anyway

**Comment by a-kriya** [comment]
Which is okay since Bulma doesn't do it that way. It would be a fairly large deviation from Bulma if Buefy were to implement dark modes for individual components like Vuetify. If this ticket is marked as a feature request, is that something you're considering?

**Comment by jtommy** [comment]
No, probably when bulma is going to release CSS variables feature, do you know other way?

**Comment by a-kriya** [comment]
Nope. Either way, it's dependent on Bulma. I'm trying to see what can be done under this ticket.

**Comment by a-kriya** [comment]
Basically, would it help clean up tickets here by direct users to Bulma's issues section if a feature/fix must come from there?

**Comment by jtommy** [comment]
Is there an open issue?

**Comment by a-kriya** [comment]
Not exactly, but jgthms/bulma#2342 along with jgthms/bulma#1837 would be close. Regardless, if there are tickets here which cannot really be addressed without underlying support from Bulma, should they not be moved to that repo?

**Comment by mapellidario** [comment]
Hello!

I saw in jgthms/bulma#2342 a recent comment by jgthms: https://github.com/jgthms/bulma/issues/2342#issuecomment-611448860 .

Would that help you closing this issue?

(anyway, kudos for all the good work :heart: :heart: )

**Comment by jtommy** [comment]
@mapellidario all depends on Bulma otherwise I should add an external to provide it. What would you do ?

**Comment by jtommy** [comment]
Waiting an official way, take a look at https://buefy.org/extensions/bulmacssvars

**Comment by Tofandel** [comment]
My PR would solve the issue of individual component themes and toggling by a simple data attribute https://github.com/jgthms/bulma/pull/2981
So adding the reference for tracking

**Comment by BroJenuel** [comment]
ops, its 2021 and dark mode still not added?

**Comment by tassoman** [comment]
Be careful, 2020 isn't finished yet :smirk: 
I've accomplished easily using one of the [Bulmaswatch](https://jenil.github.io/bulmaswatch/) themes

**Comment by BroJenuel** [comment]
> Be careful, 2020 isn't finished yet 😏
> I've accomplished easily using one of the [Bulmaswatch](https://jenil.github.io/bulmaswatch/) themes

Oh nice, I about to create a project using Buefy, hoping I can use it In the future. :)

**Comment by joaopedroAnelli** [comment]
I did this in my project using [bulma css vars](https://www.npmjs.com/package/bulma-css-vars) and [Nuxt color mode](https://color-mode.nuxtjs.org). It's a workaround, of course, but it reached my goal while we don't have dark mode natively in buefy.

**Comment by vikiival** [comment]
@jtommy is there any progress with this issue? 
Or a simple tutorial would be amazing.

**Comment by jtommy** [comment]
No, the only available options are already discussed in the previous messages.
It shouldn't be a Buefy feature since the major part of CSS comes from Bulma

**Comment by joeyboey** [comment]
> I did this in my project using [bulma css vars](https://www.npmjs.com/package/bulma-css-vars) and [Nuxt color mode](https://color-mode.nuxtjs.org). It's a workaround, of course, but it reached my goal while we don't have dark mode natively in buefy.

@joaopedroAnelli could you share your config and sass files? I can't get it to work. `npm run bulma-css-vars` does not even find the js file...

**Comment by Wakeful-Cloud** [comment]
@joeyboey
## Colors
Here are the colors I used to make [wakefulcloud.me](https://wakefulcloud.me) (Which is built with Buefy):
Name | Dark Mode Color | Light Mode Color
--- | --- | ---
`scheme-main` | `#1a1a1a` | `#ffffff`
`scheme-invert` | `#000000` | `#0a0a0a`
`text` | `#b5b5b5` | `#4a4a4a`
`text-invert` | `#c9c9c9` | `#ffffff`

## Setup
1. Create a file named `bulma-css-vars.config.js` (in the root of your project) with the below as content:
```javascript
//Must be CommonJS export
module.exports = {
  colorDefs: { //This will default to dark mode (Update with light mode colors to default to light mode)
    'scheme-main': '#1a1a1a',
    'scheme-invert': '#000000',
    'text': '#b5b5b5',
    'text-invert': '#c9c9c9'
  },
  cssFallbackOutputFile: './src/bulma-generated/generated-fallbacks.css',
  jsOutputFile: './src/bulma-generated/bulma-colors.js', //Change to ".ts" to make output TypeScript
  sassEntryFile: './src/style/main-sass-file.sass', //This file needs to exist before you run the bulma-css-vars command
  sassOutputFile: './src/bulma-generated/generated-vars.sass',
  transition: '0.5s ease'
};
```

2. Create a file named `main-sass-file.sass` in the `src/style` directory with the below as content:
```scss
//Import order matters
@import './bulma-generated/generated-fallback.css';
@import './bulma-generated/generated-vars.sass';
@import '../node_modules/bulma-css-vars/bulma-cv-lib';
```

3. Run the update script from the root of your project (Without any arguments):
```bash
node ./node_modules/bulma-css-vars/bin/bulma-css-vars.js
```

4. To invert the theme, instantiate a `ColorUpdater` and update the colors with the colors opposite of the current theme from the above table (See the TypeScript example in [Bulma CSS Vars](https://github.com/wtho/bulma-css-vars#usage) for more information)

A few very important things:
* You must use CommonJS exports for `bulma-css-vars.config.js` (`module.exports`)
* Some Bulma variables aren't supported (See [#20](https://github.com/wtho/bulma-css-vars/issues/20))
* `sassEntryFile` must exist before you run the bulma-css-vars command
* The import order of `sassEntryFile` is imporant

**Comment by joeyboey** [comment]
@Wakeful-Cloud 

Thanks for your detailed answer, I have one problem however. If I also set the primary color in the `bplma-css-vars.config.js` the color is only applied for a few components, mainly buttons and dropdowns.

But it is not applied for sliders and a lot of other ui elements and selections etc.

Do you have any advice on this?

```js
// Must be CommonJS export
module.exports = {
  // This will default to dark mode (Update with light mode colors to default to light mode)
  colorDefs: {
    primary: '#0d5d6a',
    'primary-light': '#4da9b8',
    'primary-dark': '#091a28',
    'scheme-main': '#1a1a1a',
    'scheme-invert': '#000000',
    text: '#b5b5b5',
    'text-invert': '#c9c9c9',
  },
  cssFallbackOutputFile: './bulma-generated/generated-fallback.css',
  jsOutputFile: './bulma-generated/bulma-colors.ts', // Change to ".ts" to make output TypeScript
  sassEntryFile: './assets/main.scss', // This file needs to exist before you run the bulma-css-vars command
  sassOutputFile: './bulma-generated/generated-vars.sass',
  transition: '0.5s ease',
  options: { implementation: require('sass') },
}
```

**Comment by Wakeful-Cloud** [comment]
@joeyboey Try setting the primary colors in the entry SCSS file (`./assets/main.scss`) **instead** (Above the imports from step 2):
```scss
//Customize global theme
$primary: #0d5d6a;
$primary-light: #4da9b8;
$primary-dark: #091a28;

//Import order matters
@import './bulma-generated/generated-fallback.css';
@import './bulma-generated/generated-vars.sass';
@import '../node_modules/bulma-css-vars/bulma-cv-lib';
```

This should work (It's what I did for [wakefulcloud.me](https://wakefulcloud# .me)), but, you won't be able to dynamically change the primary color anymore. You could also try identifying the exact (primary) color of the components you'd like to change using your browser's inspector, convert that color from hex to hsl, then search for it on [Bulma's variable documentation](https://bulma.io/documentation/customize/variables/), then finally add whatever variable you found to `bulma-css-vars.config.js` (This is how I figured out what variables I needed to change for the dark theme).

**Comment by joeyboey** [comment]
@Wakeful-Cloud 

Thanks a lot for your tips again. I had the order flipped..
But the problem strangely persists. Both the steps and button component use `background-color` as their `is-primary` attribute. Button works just fine, step component stays in the buefy default `rgb(121, 87, 213)`

**Comment by Wakeful-Cloud** [comment]
@joeyboey
1. Make sure you rebuild `bulma-css-vars` (Otherwise `bulma-css-vars` and `./assets/main.scss` will have colliding variable definitions).
2. Try removing `primary-light` and `primary-dark`
3. Try without `bulma-css-vars` and see if the step component changed (Though it'll be light mode). If this works, then gradually reintroduce `bulma-css-vars` and see where it breaks.

If none of the above work, you may want to open an issue with Bulma itself.

**Comment by TheJaredWilcurt** [comment]
The lack of a dark mode button to toggle the docs is what made me not use this library.

**Comment by mary-josephine-1** [comment]
Hi, any update on this?
